src/MongoDB.Driver.Core/Core/Operations/BulkWriteBatchResult.cs

```diff
@@ -45,7 +45,7 @@ internal class BulkWriteBatchResult
             var unprocessedRequests = CreateUnprocessedRequests(requests, writeErrors, isOrdered);
             var upserts = CreateUpserts(writeCommandResponse);
 
-            var n = writeCommandResponse.GetValue("n", 0).ToInt64();
+            var n = writeCommandResponse == null ? 0 : writeCommandResponse.GetValue("n", 0).ToInt64();
```

Comment:
For unacknowledged writes writeCommandResponse will be null so we have to protect against that.